crates/bitwarden-fido/src/types.rs

```diff
@@ -24,6 +24,7 @@ pub struct Fido2CredentialAutofillView {
     pub rp_id: String,
     pub user_name_for_ui: Option<String>,
     pub user_handle: Vec<u8>,
+    pub has_counter: bool,
```

Comment:
```suggestion
    /// Indicates if this credential uses a signature counter (legacy passkeys).
    /// When true, mobile clients must sync before authentication to ensure
    /// counter values are current. Modern passkeys (counter = 0) can work offline.
    pub has_counter: bool,
```
I realized while I was going over this again that I didn't understand _why_ mobile needed this counter to decide if sync was needed or not in order to support old passkeys. Here's a quote from Anders:

>That was likely the case for passkeys created before 2024-ish, but should not be the case now.
>The reason was because we implemented support for signatureCounter which required us to sync the vault on each passkey assertion.
>We’ve since tweaked our behavior and all ”modern” passkeys should not require vault access.

So, in `mobile` we're complicating the sync-by-default approach, and want to check if the field is used for a given passkey to determine if it's an "old passkey" and needs special treatment (a sync). This is some in-the-weeds Passkeys stuff, and is information that will be lost to time. I think adding a comment would be helpful.

Reply:
Fixed in [778d1f3](https://github.com/bitwarden/sdk-internal/pull/421/commits/778d1f367ce0c861255ad3fce4f76d57a421aad4)!